### PR TITLE
Fix `qml.sample` for shot vectors

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Previously `qml.sample` produced incorrect results when shot vectors are used:
```python3
dev = qml.device("default.qubit", wires=2, shots=(5, 10))

@qml.qnode(dev)
def circuit():
    qml.Hadamard(wires=0)
    qml.CNOT(wires=[0, 1])
    return qml.sample(wires=[0, 1])
```
```
>>> circuit()
(tensor([[0, 0],
        [0, 1],
        [0, 1],
        [0, 1],
        [0, 1]], dtype=int64, requires_grad=True), tensor([[1, 0],
        [1, 0],
        [1, 0],
        [1, 0],
        [0, 0],
        [0, 0],
        [1, 0],
        [1, 0],
        [1, 1],
        [1, 1]], dtype=int64, requires_grad=True))
```

**Description of the Change:**
The reshape logic in the qubit device was missing a transpose. This PR adds the transpose. Also fixed a broken test that would have failed on master.

**Related GitHub Issues:**
https://github.com/PennyLaneAI/pennylane/issues/3412

### CODE REVIEW

**Review by Qottmann — APPROVED** [review]
Good catch @eddddddy !

**Review by AlbertMitjans — APPROVED** [review]
Nice catch! Thank you for applying the solution in the CMP too 👍

File: pennylane/measurements/sample.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def process_samples(
 
         if self.obs is None:
             # if no observable was provided then return the raw samples
-            return samples if bin_size is None else samples.reshape(num_wires, bin_size, -1)
+            return samples if bin_size is None else samples.T.reshape(num_wires, bin_size, -1)
```
Comment (AlbertMitjans) [inline pennylane/measurements/sample.py:144]: I really don't like changing the `num_wires` axis here. @eddddddy would you like to check if it's possible to change this behaviour such that the `num_wires` axis is always the last axis?

It is not urgent, but it would simplify a LOT all the logic that use sampling results.

Comment (eddddddy) [reply pennylane/measurements/sample.py:144]: The device has its own logic that swaps some axes, so the output of the QNode is the expected shape. The issue is that this logic impacts some other components of the device as well.

I don't really know why we decided to do things this way but changing it right now would be out-of-scope I think.
